ESP_ATMod/command.cpp

```diff
@@ -1784,263 +2093,196 @@ void cmd_AT_CIPRECVMODE()
 }
 
 /*
- * AT+CIPRECVLEN - Get TCP Data Length in Passive Receive Mode
+ * AT+CIPSSLCERTMAX - Get or set the maximum certificates amount
  */
-void cmd_AT_CIPRECVLEN()
+void cmd_AT_CIPSSLCERTMAX()
 {
-	if (inputBuffer[13] == '?' && inputBufferCnt == 16)
+	if (inputBuffer[16] == '?' && inputBufferCnt == 19)
 	{
-		Serial.print(F("+CIPRECVLEN:"));
-
-		for (uint8_t i = 0; i <= 4; ++i)
-		{
-			int avail = 0;
-
-			if (i > 0)
-				Serial.print(',');
-
-			if (clients[i].client != nullptr)
-			{
-				avail = clients[i].client->available();
-			}
-
-			Serial.print(avail);
-		}
-
-		Serial.println();
-		Serial.printf_P(MSG_OK);
+		Serial.printf_P(PSTR("+CIPSSLCERTMAX:%d\r\nOK\r\n"), maximumCertificates);
 	}
-}
-
-/*
- * AT+CIPRECVDATA - Get TCP Data in Passive Receive Mode
- */
-void cmd_AT_CIPRECVDATA()
-{
-	uint8_t error = 1;
-
-	do
+	else if (inputBuffer[16] == '=')
 	{
-		uint8_t linkId = 0;
-		uint16_t offset;
-		uint32_t size = 0;
-
-		// Test the input
-
-		if (inputBuffer[14] != '=')
-			break;
+		uint16_t offset = 17;
+		uint32_t max;
 
-		// Read linkId
-		if (inputBuffer[15] >= '0' && inputBuffer[15] <= '5' && inputBuffer[16] == ',')
+		if (readNumber(inputBuffer, offset, max))
 		{
-			if (gsCipMux == 0)
-			{
-				Serial.println("MUX=0");
-				break;
-			}
-
-			linkId = inputBuffer[15] - '0';
+			maximumCertificates = max;
+			Settings::setMaximumCertificates(maximumCertificates);
 
-			offset = 17;
+			Serial.printf_P(MSG_OK);
 		}
 		else
-			offset = 15;
-
-		client_t *cli = &(clients[linkId]);
-
-		// Test the link
-		if (cli->client == nullptr)
-		{
-			Serial.println(F("link is not valid"));
-			break;
-		}
-
-		if (!readNumber(inputBuffer, offset, size) || offset + 2 != inputBufferCnt)
-			break;
-
-		if (size > 2048)
-		{
-			Serial.println(F("too long"));
-			break;
-		}
-
-		AT_DEBUG_PRINTF("--- linkId: %d, size: %d\r\n", linkId, size);
-
-		// Send the data
-
-		int bytes = SendData(linkId, size);
-
-		if (bytes > 0)
 		{
-			cli->lastAvailableBytes -= bytes;
-			error = 0;
-
-			// Check if the recv mode changed to 0. If so, then close the client.
-			if (gsCipRecvMode == 0)
-			{
-				DeleteClient(linkId);
-			}
+			Serial.printf_P(MSG_ERROR);
 		}
-
-	} while (0);
-
-	if (error > 0)
-		Serial.printf_P(MSG_ERROR);
+	}
 	else
-		Serial.printf_P(MSG_OK);
+	{
+		Serial.printf_P(MSG_ERROR);
+	}
 }
 
 /*
- * AT+CIPDNS - Sets User-defined DNS Servers
+ * AT+CIPSSLCERT - Load CA certificate in PEM format
  */
-void cmd_AT_CIPDNS(commands_t cmd)
+void cmd_AT_CIPSSLCERT()
 {
-	uint16_t offset = 9;
+	uint16_t offset = 13; // offset to ? or =
 
-	if (cmd != CMD_AT_CIPDNS)
-		offset += 4;
-
-	if (inputBuffer[offset] == '?' && inputBufferCnt == offset + 3)
+	if (inputBufferCnt == 15)
 	{
-		dnsConfig_t cfg;
+		PemCertificate = new char[MAX_PEM_CERT_LENGTH];

```

Comment:
Should not be called if `(CAcert.getCount() >= maximumCertificates)`, otherwise it is not deleted and causes memory leaks.

Reply:
Fixed this one.